src/media/js/app_list.js

```diff
@@ -1,6 +1,49 @@
-define('app_list', ['z'], function(z) {
-    // Remove paginated class from app lists if .loadmore goes away.
+define('app_list',
+    ['capabilities', 'storage', 'tracking', 'utils', 'z'],
+    function(caps, storage, tracking, utils, z) {
+    'use strict';
+
+    // If we've set this value in localStorage before, then always use it.
+    var expand = !!storage.getItem('expand-listings');
+    if (expand === null) {
+        // Default to the graphical view at desktop widths and traditional
+        // list view at lesser widths.
+        expand = caps.widescreen();
+    }
+
+    function setTrays(expanded) {
+        // Handle tray state and populate preview thumbs.
+        // Preserve the tray expand state in localStorage.
+        if (expanded !== undefined) {
+            expand = expanded;
+        }
+        $('.app-list').toggleClass('expanded', expanded);
+        $('.app-list-filters-expand-toggle').toggleClass('active', expand);
+        storage.setItem('expand-listings', !!expanded);
+        if (expanded) {
+            z.page.trigger('populatetray');
+            // Set the `src` for hidden images so they get loaded.
+            $('.mkt-tile img[data-src]:not([src])').each(function() {
+                this.src = this.getAttribute('data-src');
+            });
+        }
+    }
+
+    z.body.on('click', '.app-list-filters-expand-toggle', utils._pd(function() {
+        expand = !expand;
+        setTrays(expand);
+        z.doc.trigger('scroll');  // For defer image loading.
+    }));
+
+    z.page.on('loaded reloaded_chrome', function() {
+        // On load - set the tray state on available app lists.
+        if ($('.app-list').length) {
+            setTrays(expand);
```

Comment:
hm, this way when i change between app list pages, i can see the toggle changing after it loads.

i think i prefer if all the classes were set "synchronously" when the initial template is render so the user doesn't see that. of course, that is silly to do every single time we make an ```app-list``` so you think we should leave it until we get a ```<mkt-app-list>``` or something?